lib/puppet/pops/types/type_parser.rb

```diff
@@ -140,6 +140,7 @@ def self.type_map
         'iterable'     => TypeFactory.iterable,
         'iterator'     => TypeFactory.iterator,
         'string'       => TypeFactory.string,
+        'sensitive'    => TypeFactory.sensitive,
```

Comment:
That is fine. You probably do not want to accept a string instead of a type. For example the type `Sensitive['herp derp']` does not make sense. 
(You may have gleaned at the `Optional` type, which allows a string as a special thing when `Optional` is used to describe entries in a `Struct`. If you want something like that to work, more things are needed elsewhere as that is truly special.

So, drop the `unless param.is_a?(String)` bit.